master/internal/db/postgres_tasks.go

```diff
@@ -99,20 +99,14 @@ VALUES (:task_id, :allocation_id, :slots, :resource_pool, :agent_label, :start_t
 // CompleteAllocation persists the end of an allocation lifetime.
 func (db *PgDB) CompleteAllocation(a *model.Allocation) error {
 	if a.StartTime == nil {
-		if _, err := db.sql.Exec(`
-		UPDATE allocations
-		SET start_time = $2
-		WHERE allocation_id = $1`, a.AllocationID, a.EndTime); err != nil {
-			return errors.Wrap(err,
-				"setting start time to cluster heartbeat when it's assigned to zero value")
-		}
+		a.StartTime = a.EndTime
 	}
 
-	err := db.namedExecOne(`
-UPDATE allocations
-SET end_time = :end_time
-WHERE allocation_id = :allocation_id
-`, a)
+	_, err := db.sql.Exec(`
```

Comment:
the diff between this and the former version is that we used to make sure w only affected 1 row. maybe leave that check in (nice to know it happens somewhere, even those it _should_ never be broken).

Reply:
But we're using allocation_id here so that means we'll only affect one row?
Leave the check in as in check for a.startTime == nil? Then'll I'll have to have two separate queries like before?

Reply:
> the diff between this and the former version is that we used to make sure w only affected 1 row. maybe leave that check in (nice to know it happens somewhere, even those it _should_ never be broken).

But we're using allocation_id here so that means we'll only affect one row?
Leave the check in as in check for a.startTime == nil? Then'll I'll have to have two separate queries like before?

Reply:
no i mean leave the check that we only affected one row.

> But we're using allocation_id here so that means we'll only affect one row?

true yeah, i was being extra cautious since id is generated by us but i think there is a unique constraint on id so it's ok.